### DISCUSSION TITLE
THREDDS Alternatives?

### DISCUSSION THREAD

**Discussion by notstarboard** [StormSurgeLive/asgs#1075] [Q&A] [answered]
Hi All, one more discussion topic to add to the pile! There is currently no THREDDS server for Unity. The admin I've talked to also isn't very familiar with THREDDS and doesn't know what the setup would entail. I could forward the instructions on ucar.edu anyway, but before I push any more on that I wanted to make sure I understand my options better. So, have you worked with any groups who have handled distributing the results of their forecasts differently?

Per Chris Damon, who is working to visualize forecasts results on the RICHAMP project, they would just need public url(s) for the data. THREDDS is of course great for that, but I also don't want to push for it out of ignorance if there's a simpler option to consider that the admins may prefer.

**Comment by jasonfleming** [comment]
Hey Josh that is a good question that depends on your goals. We started using THREDDS about 10 years ago because it has a lot of features and it was commonly used at that time to serve oceanography and meteorology data. However, since then we have found that we don't use even a fraction of the features available, so as @wwlwpd would say, for us it is basically one step above an anonymous ftp. :-) 

Also, THREDDS is written in Java and runs inside a Tomcat container, which may seem unapproachable to younger sysadmins who may be expecting solutions for things like this to be based on node.js and javascript. However, the complexity of Tomcat etc is mitigated to some extent now that THREDDS can now be deployed via Docker. 

One other thing is that (technically) we don't use THREDDS to distribute results; it is really middleware that only exposes results so they can be downloaded. As a result, some kind of discovery or notification mechanism is still needed to let data consumers know that (a) results are available, (b) what they consist of, and (c) the URL where they are located. So, dissemination of results is really a combination of situational awareness and active/selective downloading by data consumers.

In any case, I think the simplest possible option would be to set up a machine with a lot of disk space and an ssh server that both you and RICHAMP can access. You could post files there from ASGS via `scp` using `opendap_post2.sh` (the script is misnamed because it actually has nothing to do with opendap, it just copies files to an external server). Automated emails from ASGS would be used to maintain situational awareness. When data consumers see that results have been posted that they are interested in, they could download them from the shared server via `scp`.

**Comment by wwlwpd** [comment, marked as answer]
The only non-traditional alterntative we've considered is https://min.io/ (minio) which basically implements a private S3 server; it's easy to set up as a Docker container. But if you just need public URLs, just set up a simple static http server (nginx) and make sure the directory listing is turned on. FTP is also not a terrible idea, but it's just another service to deal with.

**Comment by notstarboard** [comment]
I appreciate the information and both of your suggestions! I'll run these by the Unity admin I've been working with and get his thoughts on which option he thinks would be best.
